openthread: Add support for platform message management (#18564)
nrfconnect/sdk-nrf: doc/nrf/protocols/thread/overview/ot_memory.rst

This page provides information about the layout and the amount of flash memory and RAM that is required by samples that use the OpenThread stack.
Use these values to see whether your application has enough space for running on particular platforms for different application configurations.

> .. _thread_ot_memory_configuration:
> 
> Message pool configuration
> **************************
> 
> OpenThread uses a message pool to manage memory for message buffers.
> Memory for the message pool can be statically allocated by the OpenThread stack or managed by the platform.
> The :kconfig:option:`CONFIG_OPENTHREAD_PLATFORM_MESSAGE_MANAGEMENT` Kconfig option can be used to
> enable platform message management.
> 
> Message buffer size and number of message buffers in the pool can be configured with :kconfig:option:`CONFIG_OPENTHREAD_MESSAGE_BUFFER_SIZE`
> and :kconfig:option:`CONFIG_OPENTHREAD_NUM_MESSAGE_BUFFERS` Kconfig options, respecitvely. By
> default, both options are set to ``128``.
> 
> .. important::
> 
>    Please note, that when using :ref:`thread_ug_prebuilt`, changing :kconfig:option:`CONFIG_OPENTHREAD_PLATFORM_MESSAGE_MANAGEMENT`
>    will have no effect. Additionally, :kconfig:option:`CONFIG_OPENTHREAD_MESSAGE_BUFFER_SIZE` Kconfig option have to be set
>    to the same value, as in the pre-built library.
> 

Reviewer: I would move this section to `doc/nrf/protocols/thread/configuring.rst` instead, under the Additional configuration options heading, after Thread Specification options.
```suggestion
.. _thread_configuring_messagepool:

Message pool configuration
**************************

OpenThread uses a message pool to manage memory for message buffers.
Memory for the message pool can be statically allocated by the OpenThread stack or managed by the platform.
You can use the :kconfig:option:`CONFIG_OPENTHREAD_PLATFORM_MESSAGE_MANAGEMENT` Kconfig option to enable platform message management.

Message buffer size and number of message buffers in the pool are be configured with the :kconfig:option:`CONFIG_OPENTHREAD_MESSAGE_BUFFER_SIZE` and :kconfig:option:`CONFIG_OPENTHREAD_NUM_MESSAGE_BUFFERS` Kconfig options, respectively.
By default, both options are set to ``128``.

.. note::

   When using :ref:`thread_ug_prebuilt`, changing the :kconfig:option:`CONFIG_OPENTHREAD_PLATFORM_MESSAGE_MANAGEMENT` Kconfig option will have no effect.
   Additionally, the :kconfig:option:`CONFIG_OPENTHREAD_MESSAGE_BUFFER_SIZE` Kconfig option has  to be set to the same value that is used in the pre-built library.
```
